### ISSUE TITLE
BUG : Different value at the report of roads inundation (Flood and Tsunami)

### ISSUE THREAD

**Issue by samnawi** [inasafe/inasafe#2361]
# Problem
There is a different value that appear in the report, between in the "Breakdown by road type" and in the "Detailed road type report".
![selection_197](https://cloud.githubusercontent.com/assets/8871958/9866967/46900a76-5b93-11e5-8487-245cbef25317.png)

# Data
Hazard : Tsunami Maumere
               Jakarta Flood Polygon (RW Boundary)
               Jakarta Flood Raster (HKV - Modeling)
Exposure : OSM Roads

# Proposed Solution
Make it happy

# CC
@Charlotte-Morgan @timlinux 

Thank you

**Comment by timlinux** [comment]
This is because the road types list is hard coded in the post processor file - see https://github.com/AIFDR/inasafe/blob/develop/safe/postprocessors/road_type_postprocessor.py#L41

**Comment by timlinux** [comment]
I have marked this as for 3.3 as fixing it is going to cause some major code churn.

**Comment by samnawi** [comment]
Ok @timlinux.
When I read your comment, it's problem because the road types list is hard coded, I curious about it. Then I try to run again using national data from BIG (Indonesia Geospatial Agency), apparently InaSAFE could not compute it. See #2369.

Is the problem also related about the classification in this link https://github.com/AIFDR/inasafe/blob/develop/safe/postprocessors/road_type_postprocessor.py#L41 ?

Because I read in line 42 - 54, mention about the road types that I think it's similar with OSM roads classification. So, InaSAFE can not read the other classification?

Thank you.

**Comment by timlinux** [comment]
Hi @samnawi correct - that road_type_postprocessor contains the hard coded list I was referring to. I am working on a fix for this.

**Comment by raniedwianugrah** [comment]
Hi @timlinux 

The problem still exist in InaSAFE 3.2.5, there are differences the total length of the affected road between "Analysis Result" column and "Closed Roads" column in statistic table result. 

![table_roads](https://cloud.githubusercontent.com/assets/2528195/11947588/d7d2302c-a89b-11e5-8495-e683c91f9907.PNG)

![roads table](https://cloud.githubusercontent.com/assets/2528195/11947592/e9da84ea-a89b-11e5-9578-831ab456de48.PNG)

The data used are
- Jakarta_Flood_18113_WGS84.shp
- Jakarta_Flood_HKV_WGS84.tif
- Roads from OSM Downloader
- Jakarta_Subdistrict_WGS84.shp

CC : @adissadis @Charlotte-Morgan @samnawi

**Comment by adissadis** [comment]
Hi @timlinux @Charlotte-Morgan @samnawi 

This problem still appear in InaSAFE 3.3 dev master. Also in detailed result it still named as "Affected" while in general result and breakdown result it said "temporary closed".
![image](https://cloud.githubusercontent.com/assets/2520159/12258457/7ad30a8e-b941-11e5-948a-1223074afabe.png)

![image](https://cloud.githubusercontent.com/assets/2520159/12258500/b9170002-b941-11e5-9bf5-57452c3a76bf.png)

**Comment by adissadis** [comment]
I forget to mention this:
![image](https://cloud.githubusercontent.com/assets/2520159/12258598/689b2f80-b942-11e5-9297-4652c29cffa4.png)

Total road affected have same number with Other

**Comment by Charlotte-Morgan** [comment]
@samnawi - can you please look at this and see if it can be closed?

**Comment by samnawi** [comment]
@adissadis I think because your road classification just "other". So, the total of affected road have same number with "other".
 @Charlotte-Morgan I think it's already good for Flood and Tsunami raster, but I still have a few question for flood vector VS roads, as explain in this picture.
![selection_089](https://cloud.githubusercontent.com/assets/8871958/13663210/7c39a892-e6d2-11e5-80a4-4dc321c15fd0.png)
Is my assumption in that picture is correct?

**Comment by Charlotte-Morgan** [comment]
@samnawi - thanks for the update
It looks like the IF and post processing are counding different features and different units.
The IF is reporting meters for all road types
the post processing is counting number of features and only some features.

This needs work as we integrate the aggregation into the IF - @Gustry

**Comment by timlinux** [comment]
From my side I took a look at the roads aggregator. It inherits from buildings and thus is doing feature counts for its aggregation. I discussed this with @Charlotte-Morgan - we think it make far more sense to rather report on length(m) of each road type affected. This I propose we:

Move some of the logic in ``safe/postprocessors/building_type_postprocessor.py`` into ``safe/postprocessors/road_type_postprocessor.py`` and then measure the length in m of each segment and report the total length of each type affected in the aggregator.

@Gustry
